tools/build-tools/src/test/typeValidator/classValidator.spec.ts

```diff
@@ -49,92 +49,503 @@ describe("Class", () => {
+            interface TestInterface {}
+            export class TestClass implements TestInterface {}
+            `;
+
+            const classNew =
+            `
+            interface TestInterface2 {}
+            export class TestClass implements TestInterface2 {}
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // reordering implementations is not breaking
+        it("reorders changed class implementations", () => {
+            const classOld =
+            `
+            interface TestInterface1 {}
+            interface TestInterface2 {}
+            export class TestClass implements TestInterface1, TestInterface2 {}
+            `;
+
+            const classNew =
+            `
+            interface TestInterface1 {}
+            interface TestInterface2 {}
+            export class TestClass implements TestInterface2, TestInterface1 {}
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.none);
+        });
+    });
+
+    describe("type parameters", () => {
+        // adding/removing class type parameters is not a breaking change if they
+        // aren't also exposed on the public API elsewhere
+        it("adds a class type parameters not exposed on the public API", () => {
+            const classOld =
+            `
+            export class TestClass {
+                private prop1?: Array<string>;
+            }
+            `;
+
+            const classNew =
+            `
+            export class TestClass<X> {
+                private prop1?: Array<X>;
+            }
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.none);
+        });
+
+        // class type parameter changes exposed on the public API are the same as
+        // equivalent typing changes
+        it.skip("changes a class type parameters exposed on the public API", () => {
+            const classOld =
+            `
+            export class TestClass {
+                public prop1?: Array<string>;
+            }
+            `;
+
+            const classNew =
+            `
+            export class TestClass<X> {
+                public prop1?: Array<X>;
+            }
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // changing type parameter names is not breaking
+        it("renames class type parameters", () => {
+            const classOld =
+            `
+            export class TestClass<X, Y> {
+                private prop1?: Array<X>;
+                private prop2?: Array<Y>;
+            }
+            `;
+
+            const classNew =
+            `
+            export class TestClass<Y, X> {
```

Comment:
nit: maybe using completely different names in the new class would make it clearer that you're not testing reordering of the type params.